File: Technology Fundamentals/02.Exercises - Intro and Basic Syntax/Java/src/VendingMachine.java
import java.util.*;

public class VendingMachine {
    static boolean contains(double[] arr, double item){
        for(int i =0; i < arr.length; i++){
            if (arr[i] == item){
                return true;
            }
        }
        return false;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String input = scanner.nextLine();

        double totalCoins = 0;
        double[] validCoins = new double[]{
                0.1, 0.2, 0.5, 1, 2
        };

        Map<String, Double> validProducts = new HashMap<>();

        validProducts.put("Nuts", 2.0);
        validProducts.put("Water", 0.7);
        validProducts.put("Crisps",1.5);
        validProducts.put("Soda",0.8);
        validProducts.put("Coke",1.0);

        while (!Objects.equals(input, "Start"))
        {
            double coins = Double.parseDouble(input);
            if (contains(validCoins, coins))
            {
                totalCoins += coins;
            }
            else
            {
                System.out.printf("Cannot accept %s\n", coins);
            }

            input = scanner.nextLine();
        }

        String product = scanner.nextLine();

        while (true)
        {
            if (Objects.equals(product, "End"))
            {
                break;
            }

            if (validProducts.containsKey(product))
            {
                double value = validProducts.get(product);
                if ((totalCoins - value) >= 0)
                {
                    System.out.printf("Purchased %s\n", product);
                    totalCoins -= value;
                }
                else
                {
                    System.out.println("Sorry, not enough money");
                }

            }
            else
            {
                System.out.println("Invalid product");
            }

            product = scanner.nextLine();
        }

        System.out.printf("Change: %.2f\n", totalCoins);
    }
}

File: C# Fundamentals May 2019/02.Exercises - Intro and Basic Syntax/PHP/01.Exercises/VendingMachine.php
<?php
$sumCoins = 0.0;

while (true) {
    $coin = readline();

    if ($coin === "Start") {
        break;
    }

    $coinNumber = floatval($coin);

    switch ($coinNumber) {
        case 0.1:
        case 0.2:
        case 0.5:
        case 1:
        case 2:
            $sumCoins += $coinNumber;
            break;
        default:
            echo "Cannot accept $coin".PHP_EOL;
            break;
    }
}
$priceProduct = 0;
while (true) {
    $product = readline();

    if ($product === "End") {
        break;
    }

    $validProduct = true;
    switch ($product) {
        case "Nuts":
            $priceProduct = 2.0;
            $validProduct = true;
            break;
        case "Water":
            $priceProduct = 0.7;
            $validProduct = true;
            break;
        case "Crisps":
            $priceProduct = 1.5;
            $validProduct = true;
            break;
        case "Soda":
            $priceProduct = 0.8;
            $validProduct = true;
            break;
        case "Coke":
            $validProduct = true;
            $priceProduct = 1.0;
            break;
        default:
            echo "Invalid product".PHP_EOL;
            $validProduct = false;
            break;
    }

    if ($validProduct) {
        if ($priceProduct <= $sumCoins) {
            $toLower = strtolower($product);
            echo "Purchased $toLower".PHP_EOL;
            $sumCoins -= $priceProduct;
        } else {
            echo "Sorry, not enough money".PHP_EOL;
        }
    }
}
$result = number_format($sumCoins, 2, '.', '');
echo "Change: $result";
?>
